lib/nm/source.rb

```diff
@@ -28,15 +29,22 @@ def data(file_path)
     end
 
     def render(template_name, locals = nil)
-      @template_class.new(self, source_file_path(template_name), locals || {}).__data__
+      if (filename = source_file_path(template_name)).nil?
+        template_desc = "a template named #{template_name.inspect}"
+        if !@ext.nil? && !template_name.end_with?(@ext)
+          template_desc += " with #{@ext.inspect} as an extension"
+        end
+        raise ArgumentError, "#{template_desc} does not exist"
+      end
```

Comment:
@jcredding here I add checking that a filename exists and raising a friendly argument error if not.  I made the template description aware of any specified ext and if the given template name already ended with it.